sorc/ncep_post.fd/SURFCE.f

```diff
@@ -5141,7 +5142,15 @@ SUBROUTINE SURFCE
 !-- TOTPRCP is total 1-hour accumulated precipitation in  [m]
 !-- RAP/HRRR and RRFS use 1-h bucket. GFS uses 3-h bucket
 !-- so this section will need to be revised for GFS
-               totprcp = (AVGPREC(I,J)*3600.*1000./DTQ2)
+               IF (MODELNAME .eq. 'FV3R') THEN
+                 if(AVGPREC(I,J)/=spval)then
+                   totprcp = (AVGPREC(I,J)*3600./DTQ2)
+                 else
+                   totprcp = 0.0
+                 endif
+               ELSE
+                 totprcp = (RAINC_BUCKET(I,J) + RAINNC_BUCKET(I,J))*1.e-3
```

Comment:
@EricJames-NOAA From my testing, your new commit still change results in rap/hrrr tests.

Reply:
@WenMeng-NOAA thanks.  I guess I have a philosophical question about whether we need to preserve the calculation as carried out in the operational RAP/HRRR, or if we don't need to worry about that.  I guess we have the UPP hash used for the RAP/HRRR, so we can always go back to that.  So maybe this model check is not needed.  What are your thoughts?

Reply:
@EricJames-NOAA It would better to keep these RAP/HRRR calculation unless it is bug. Adding modelname threshold is a good way for modification for ufs-based models. It seems the changes in rap/hrrr  testing is introduced by "totprcp=(rainc_bucket(i,j)+rainne_bucket(i,j,))*1e-3"(line 5152). I would like to know it is an intentional change for rap/hrrr?

Reply:
@WenMeng-NOAA thanks.  That line is the original formulation for RAP/HRRR, which I changed in an earlier PR to implement the GSL p-type for RRFS:

https://github.com/NOAA-EMC/UPP/commit/7e58735648a58652dbd6f02b825f19a96db68e6b

So I think this code should preserve the original RAP/HRRR calculation.  I think we see differences because I had previously replaced that line with the new "totprcp = (AVGPREC(I,J)*3600./DTQ2)"

Reply:
@EricJames-NOAA Thanks for explaining. So I would think the current changes are ready for testing.